packages/flutter/lib/src/material/switch.dart

```diff
@@ -134,15 +134,21 @@ class Switch extends StatelessWidget {
   /// or macOS, following Material design's
   /// [Cross-platform guidelines](https://material.io/design/platform-guidance/cross-platform-adaptation.html).
   ///
-  /// On iOS and macOS, this constructor creates a [CupertinoSwitch], which has
-  /// matching functionality and presentation as Material switches, and are the
+  /// On iOS and macOS, this constructor creates a [Switch] which has matching
+  /// functionality and presentation as Material switches, and are the
   /// graphics expected on iOS. On other platforms, this creates a Material
   /// design [Switch].
   ///
-  /// If a [CupertinoSwitch] is created, the following parameters are ignored:
-  /// [activeTrackColor], [inactiveThumbColor], [inactiveTrackColor], [trackOutlineWidth]
-  /// [activeThumbImage], [onActiveThumbImageError], [inactiveThumbImage],
-  /// [onInactiveThumbImageError], [materialTapTargetSize].
+  /// To customize iOS/macOS switch theme, create a subclass for [Adaptation]
```

Comment:
To provide a custom switch theme that's only used by this factory constructor, add a custom `Adaptation<SwitchThemeData>` class to to [ThemeData.adaptations].  This can be useful in situations where you don't want the overall theme's [ThemeData.switchTheme] to apply when this adaptive constructor is used.